cluster/validation.go

```diff
@@ -217,6 +217,14 @@ func validateNetworkOptions(c *Cluster) error {
 	if c.Network.Plugin == FlannelNetworkPlugin && c.Network.MTU != 0 {
 		return fmt.Errorf("Network plugin [%s] does not support configuring MTU", FlannelNetworkPlugin)
 	}
+
+	if c.Network.Plugin == WeaveNetworkPlugin {
+		err := warnWeaveDeprecation(c.Version)
+		if err != nil {
+			return fmt.Errorf("Error while printing Weave deprecation message: %v", err)
+		}
```

Comment:
Nit: Combine the assignment and the if check
```suggestion
		if err := warnWeaveDeprecation(c.Version); err != nil {
			return fmt.Errorf("Error while printing Weave deprecation message: %w", err)
		}
```